Source/ExecutionEngine/ConcurrentToSequentialWriteManager.cs

```diff
@@ -0,0 +1,60 @@
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.Boogie;
+
+public class ConcurrentToSequentialWriteManager
```

Comment:
I think it might be helpful to include a comment describing in some detail what this class is supposed to do.

I found reading the test to be the easiest way to get an intuition for it. Given that intuition, I can now see how this class implements it. But before wrapping my head around that, I wasn't sure what the purpose of the code was.